src/state.rs

```diff
@@ -120,12 +125,12 @@ impl ProgressState {
     }
 
     /// The number of steps per second
-    pub fn per_sec(&self) -> u64 {
-        let avg_time = self.est.seconds_per_step();
-        if avg_time == 0.0 {
-            0
+    pub fn per_sec(&self) -> f64 {
+        let per_sec = 1.0 / self.est.seconds_per_step();
```

Comment:
Why did you change the structure like this? I think I prefer comparing to zero rather than `is_nan()`.

Reply:
Two reasons: even when comparing the denominator to zero, I sometimes get `NaN` instead of 0 as the result anyway. Probably due to using `==` on floating point values. So, semantically, I think that comparing to `NaN` is more accurate.

Other reason is that the codegen seems a bit better when comparing to `NaN`. The compiler only generates branchless code when comparing to NaN, and there are fewer instructions overall. https://godbolt.org/z/f4fhPs7ec

Reply:
Sounds good!